src/main/java/bio/terra/app/utils/TimUtils.java

```diff
@@ -34,6 +38,27 @@ public static String encode(String s) {
         return result.toString();
     }
 
+    /**
+     * Encode the AS columns of a SQL query into corresponding TIM property names
+     *
+     * @param sql the SQL string containing AS columns to encode
+     * @param columnReplacements a map which stores column names as keys and TIM property names as values
+     * @return a modified SQL string containing encoded TIM property names
+     */
+    public static String encodeSqlColumns(String sql, Map<String, String> columnReplacements) {
+        Pattern regex = Pattern.compile("( [aA][sS] )(\\w+)");
```

Comment:
What if you have AS for a table clause? Also, presuming this means that subqueries won't be supported? (e.g. ```select (select a.col2 AS col2 from tableA AS a where a.col1=b.col1)` AS c from tableB AS b```

Reply:
It's not ideal, but since we control the input, SQL, and column mapping, we can avoid having collisions. But it's also the case that the only columns that are replaced are those that are in the columnReplacements map, so we can support arbitrary queries as long as they don't use the same column names in the subqueries.

Reply:
@pshapiro4broad are going to chat about this a bit more soon.

Reply:
First, I don't think we need to do anything for subqueries; I think the mapping of BQ columns to exported (TIM) names only occurs in the top level `SELECT` clause.

One possible heuristic would be to split the query up so we first isolate the data between `SELECT` and `FROM`. Once we have that, we can apply the `AS` regex to the entire contents. If necessary, the heuristic could also skip over quoted blocks and subselect blocks.